mdp/nodes/classifier_nodes.py

```diff
@@ -209,12 +269,16 @@ def __init__(self, execute_method=None,
         self._shuffled_update = True
 
     def _get_supported_dtypes(self):
+        """
+        :return: The list of dtypes supported by this node.
+        :rtype: list
+        """
         return ['b']
 
     def _train(self, x):
         """Provide the hopfield net with the possible states.
-
-        x -- a matrix having different variables on different columns
+        
+        :param x: A matrix having different variables on different columns
```

Comment:
why not add ``:type x: numpy.ndarray``?

Reply:
IIRC I was not sure at the time whether MDP generalized to iterables. Due to limited knowledge of the specific modules, I was easy on said types. If that abstraction is not featured for this class, I will adapt the docstring. See http://mdp-toolkit.sourceforge.net/tutorial/iterables.html